awd/cli.py

```diff
@@ -152,28 +160,28 @@ def deposit(
         )
         return  # Unable to access S3 bucket, exit application
 
-    doi_table = DoiProcessAttempt()
-    doi_table.set_table_name(ctx.obj["doi_table"])
-    if not doi_table.exists():
+    doi_process_attempt = DoiProcessAttempt()
+    doi_process_attempt.set_table_name(ctx.obj["doi_table_name"])
+    if not doi_process_attempt.exists():
```

Comment:
I didn't catch this the first time around, but I think this is bit awkward here.

It looks as though you're instantiating what's basically an empty `DoiProcessAttempt` model instance and affixing it to the `Article` class.  That might make sense if a) you know it doesn't exist, and b) you build it up incrementally and then eventually save it.

But in the `Article.process()` flow, the first step is checking to see if it exist and adding a row if not.  Right then, I think you have the ability to affix that row to the Article class.

This pattern could help with the comment above about `check_doi_and_add_to_table()`.  

I think has a rule of thumb, and methods on the `DoiProcessAttempt` table that are doing things like searching rows, adding rows, etc., those should be `classmethod`.  Then only methods that truly utilize `self`, which would be an actual row from the database, would be normal methods.

Reply:
To be even more concrete, I might suggest that `Article` is _not_ initialized with an instance of `DoiProcessAttempt`.  If it's `None` to begin with -- indicating you have no tether to the dynamodb table yet -- and then see where that leads with reworking _how_ an instance is affixed to the `Article` instance.